extensions/CESIUM_batch_table_hierarchy/README.md

```diff
@@ -0,0 +1,420 @@
+{
+    "lampStrength" : [10, 5, 7, 0, 0, 0, 0, 0],
+    "lampColor" : ["yellow", "white", "white", "", "", "", "", ""],
+    "carType" : ["", "", "", "truck", "bus", "sedan", "", ""],
+    "carColor" : ["", "", "", "green", "blue", "red", "", ""],
+    "treeHeight" : [0, 0, 0, 0, 0, 0, 10, 15],
+    "treeAge" : [0, 0, 0, 0, 0, 0, 5, 8]
+}
+```
+
+In this example, several `""` and `0` array values are stored so each array has the same number of elements.  A potential workaround is to store properties as JSON objects; however, this becomes bulky as the number of properties grows:
+
+```json
+{
+  "info" : [
+    {
+      "lampStrength" : 10,
+      "lampColor" : "yellow"
+    },
+    {
+      "lampStrength" : 5,
+      "lampColor" : "white"
+    },
+    {
+      "lampStrength" : 7,
+      "lampColor" : "white"
+    },
+    {
+      "carType" : "truck",
+      "carColor" : "green"
+    },
+    {
+      "carType" : "bus",
+      "carColor" : "blue"
+    },
+    {
+      "carType" : "sedan",
+      "carColor" : "red"
+    },
+    {
+      "treeHeight" : 10,
+      "treeAge" : 5
+    },
+    {
+      "treeHeight" : 15,
+      "treeAge" : 8
+    }
+  ]
+}
+```
+
+Another limitation of the standard batch table is the difficulty in expressing metadata hierarchies. 
+For example, consider a tile that represents a city block. The block itself contains metadata, the individual buildings contain metadata, and the building walls contain metadata. A tree diagram of the hierarchy might look like this:
+
+- block
+  - building
+    - wall
+    - wall
+  - building
+    - wall
+    - wall
+  - building
+    - wall
+    - wall
+
+
+In order to select a wall and retrieve properties from its building, the wall metadata must also include building metadata. Essentially the three-level hierarchy must be flattened into each feature, resulting in a lot of duplicate entries.
+
+A standard batch table with two walls per building and three buildings per block might look like this:
+
+```json
+{
+    "wall_color" : ["blue", "pink", "green", "lime", "black", "brown"],
+    "wall_windows" : [2, 4, 4, 2, 0, 3],
+    "building_name" : ["building_0", "building_0", "building_1", "building_1", "building_2", "building_2"],
+    "building_id" : [0, 0, 1, 1, 2, 2],
+    "building_address" : ["10 Main St", "10 Main St", "12 Main St", "12 Main St", "14 Main St", "14 Main St"],
+    "block_lat_long" : [[0.12, 0.543], [0.12, 0.543], [0.12, 0.543], [0.12, 0.543], [0.12, 0.543], [0.12, 0.543]],
+    "block_district" : ["central", "central", "central", "central", "central", "central"],
+}
+```
+
+Both these cases illustrate the benefit of supporting feature types and a feature hierarchy within the Batch Table.
+
+## Batch table JSON schema updates
+
+The standard batch table may be extended to include a `CESIUM_batch_table_hierarchy` object that defines a set of classes and a tree structure for class instances.
+
+Sample Batch Table:
+
+```json
+{
+  "CESIUM_batch_table_hierarchy" : {
```

Comment:
`CESIUM_batch_table_hierarchy` should be contained within an `extensions` objects.